source/Halibut.Tests/Support/BackwardsCompatibility/ClientAndPreviousServiceVersionBuilder.cs

```diff
@@ -71,11 +71,15 @@ public async Task<IClientAndService> Build()
                 runningOldHalibutBinary = await new HalibutTestBinaryRunner(serviceConnectionType, listenPort, clientCertAndThumbprint, serviceCertAndThumbprint, version).Run();
                 serviceUri = new Uri("poll://SQ-TENTAPOLL");
             }
-            else
+            else if (serviceConnectionType == ServiceConnectionType.Listening)
             {
                 runningOldHalibutBinary = await new HalibutTestBinaryRunner(serviceConnectionType, null, clientCertAndThumbprint, serviceCertAndThumbprint, version).Run();
                 serviceUri = new Uri("https://localhost:" + runningOldHalibutBinary.serviceListenPort);
             }
+            else
+            {
+                throw new NotSupportedException();
```

Comment:
I haven't added support here yet but wanted to make sure it blows up if someone tries to use it with PollingOverWebSockets